src/command/subgraph/check.rs

```diff
@@ -0,0 +1,140 @@
+            check::check_partial_schema_query::Variables {
+                graph_id: self.graph.name.clone(),
+                variant: self.graph.variant.clone(),
+                partial_schema,
+                implementing_service_name: self.service_name.clone(),
+            },
+            &client,
+        )
+        .context("Failed to validate schema")?;
+
+        tracing::info!(
+            "Validated the proposed subgraph against metrics from {}@{}",
+            &self.graph.name,
+            &self.graph.variant
+        );
+
+        match res {
+            check::CheckResponse::CompositionErrors(composition_errors) => {
+                handle_composition_errors(&composition_errors)
+            }
+            check::CheckResponse::CheckResult(check_result) => handle_checks(check_result),
+        }
+    }
+}
+
+fn handle_checks(check_result: check::CheckResult) -> Result<RoverStdout> {
+    let num_changes = check_result.changes.len();
+
+    let msg = match num_changes {
+        0 => "There is no difference between the proposed subgraph and the subgraph that already exists in the graph registry. Try making a change to your proposed subgraph before running this command.".to_string(),
+        _ => format!("Compared {} schema changes against {} operations", check_result.changes.len(), check_result.number_of_checked_operations),
+    };
+
+    tracing::info!("{}", &msg);
+
+    let mut num_failures = 0;
+
+    if !check_result.changes.is_empty() {
+        let mut table = Table::new();
+        table.add_row(row!["Change", "Code", "Description"]);
+        for check in check_result.changes {
+            let change = match check.severity {
+                check::check_partial_schema_query::ChangeSeverity::NOTICE => "PASS",
+                check::check_partial_schema_query::ChangeSeverity::FAILURE => {
+                    num_failures += 1;
+                    "FAIL"
+                }
+                _ => unreachable!("Unknown change severity"),
+            };
+            table.add_row(row![change, check.code, check.description]);
+        }
+
+        eprintln!("{}", table);
+    }
+
+    if let Some(url) = check_result.target_url {
+        tracing::info!("View full details here");
+        tracing::info!("{}", url.to_string());
+    }
+
+    match num_failures {
+        0 => Ok(RoverStdout::None),
+        1 => Err(anyhow::anyhow!(
+            "Encountered 1 failure while checking your subgraph."
+        )),
+        _ => Err(anyhow::anyhow!(
+            "Encountered {} failures while checking your subgraph.",
+            num_failures
+        )),
+    }
+}
+
+fn handle_composition_errors(
+    composition_errors: &[check::check_partial_schema_query::CheckPartialSchemaQueryServiceCheckPartialSchemaCompositionValidationResultErrors],
+) -> Result<RoverStdout> {
+    let mut num_failures = 0;
+    for error in composition_errors {
+        num_failures += 1;
+        tracing::error!("{}", &error.message);
```

Comment:
as a potential future improvement, we could make a table of these errors similar to checks. Composition errors should always be in the format `[Service] Type.field? -> Message` so maybe we could parse out into two columns like `location` and `message`. But this is fine for now :)

Reply:
yeah, the way it prints it is as so: 

```
ERROR [films] Person -> A @key directive specifies a field which is not found in this service. Add a field to this type with @external.
 ERROR [films] Person -> A @key selects iddd, but Person.iddd could not be found
 ERROR [films] Person -> extends from people but specifies an invalid @key directive. Valid @key directives are specified by the originating type. Available @key directives for this type are:
        @key(fields: "id")
Error: Encountered 3 composition errors while composing the subgraph
```

i think if we wanted to print it like a table we'd want to have that data returned in separate fields by the API rather than trying to parse out these strings.